front/lib/api/vaults.ts

```diff
@@ -102,12 +105,42 @@ export const getDataSourceContent = async (
 };
 
 export const getManagedDataSourceContent = async (
+  auth: Authenticator,
   connectorId: string,
-  permission: ConnectorPermission,
+  permission: ConnectorPermission | undefined,
   rootIds: string[] | null,
   parentId: string | null,
   viewType: ContentNodesViewType
 ): Promise<Result<LightContentNode[], ConnectorsAPIError>> => {
+  switch (permission) {
```

Comment:
shouldn't we do the same tests for `getDataSourceContent` ?

Reply:
(maybe it can be simplified with `if (permission === "write" && !auth.isBuilder())` and `if (!permission && !auth.isAdmin())`

Reply:
> shouldn't we do the same tests for getDataSourceContent ?

It wasn't applied outside of managed sources so we replicated the same here. In any case, it's read everywhere except for Slack channel integration (write).

Reply:
>(maybe it can be simplified with if (permission === "write" && !auth.isBuilder()) and if (!permission && !auth.isAdmin())

I find it less explicit / readable than the switch case tbh.

But I agree this code is verbose, maybe we should refactor later (or maybe it's going to be dropped ?)